src/layer/S1GRDAWSEULayer.ts

```diff
@@ -53,28 +56,41 @@ export class AbstractS1GRDAWSLayer extends AbstractSentinelHubV3Layer {
     dataProduct: string | null = null,
     title: string | null = null,
     description: string | null = null,
+    acquisitionMode: AcquisitionMode | null = null,
     polarization: Polarization | null = null,
+    resolution: Resolution | null = null,
     orthorectify: boolean | null = false,
     backscatterCoeff: BackscatterCoeff | null = BackscatterCoeff.GAMMA0_ELLIPSOID,
-    resolution: Resolution | null = null,
   ) {
     super(instanceId, layerId, evalscript, evalscriptUrl, dataProduct, title, description);
+    this.acquisitionMode = acquisitionMode;
     this.polarization = polarization;
+    this.resolution = resolution;
     this.orthorectify = orthorectify;
     this.backscatterCoeff = backscatterCoeff;
-    this.resolution = resolution;
   }
 
   protected async updateProcessingGetMapPayload(payload: ProcessingPayload): Promise<ProcessingPayload> {
-    const layerParams = await this.fetchLayerParamsFromSHServiceV3();
+    if (this.polarization === null || this.acquisitionMode === null || this.resolution === null) {
+      if (this.instanceId === null || this.layerId === null) {
+        throw new Error(
+          "Parameters polarization, acquisitionMode and/or resolution are not set, but can't fetch them from service - instanceId and layerId are not available",
```

Comment:
Maybe better phrased: 
`Parameters polarization, acquisitionMode and/or resolution are not set and can't be fetched from service because instanceId and layerId are not available.`

Reply:
Much better, thank you!

Reply:
Resolving.